src/components/secrets/Secret.tsx

```diff
@@ -208,7 +208,8 @@ export const SecretForm: React.FC<SecretFormProps> = function (props) {
     const [isSubPathChecked, setIsSubPathChecked] = useState(!!props.subPath)
     const [isFilePermissionChecked, setIsFilePermissionChecked] = useState(!!props.filePermission)
     const [filePermissionValue, setFilePermissionValue] = useState({ value: props.filePermission, error: "" })
-    const isChartVersion309OrBelow = props.appChartRef && isVersionLessThanOrEqualToTarget(props.appChartRef.version, [3, 9]);
+    const chartVersionNum = getMajorAndMinorVersionArr(props.appChartRef.version);
+    const isChartVersion309OrBelow = props.appChartRef && isVersionLessThanOrEqualToTarget(chartVersionNum, [3, 9]);
```

Comment:
can we pull out these major and minor versions and use names which are declarative, so that we know why are we using this version for eg 7 -> minorVersionBeforeAppMetricsSupport

Reply:
added constants MINCHARTVERSION_FOR_APPMETRICS_SUPPORT = [3, 7, 0] and MINCHARTVERSION_FOR_SUBPATH_SUPPORT = [3, 9]